src/main/java/de/tum/in/www1/artemis/service/SubmissionVersionService.java

```diff
@@ -80,7 +80,8 @@ private SubmissionVersion createNewVersion(Submission submission, User user) {
 
     private String getSubmissionContent(Submission submission) {
         if (submission instanceof ModelingSubmission) {
-            return ((ModelingSubmission) submission).getModel();
+            return ("Model: " + ((ModelingSubmission) submission)
+                .getModel() + "; Explanation: " + ((ModelingSubmission) submission).getExplanationText());
```

Comment:
Can you cast it to modeling submission once and use it?

Reply:
Sure. I changed it.